proto/internal/temporal/server/api/worker_deployment/v1/message.proto

```diff
@@ -0,0 +1,183 @@
+// The MIT License
+//
+// Copyright (c) 2024 Temporal Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+syntax = "proto3";
+
+package temporal.server.api.worker_deployment.v1;
+
+option go_package = "go.temporal.io/server/api/worker_deployment/v1;worker_deployment";
+
+import "temporal/api/enums/v1/task_queue.proto";
+import "google/protobuf/timestamp.proto";
+import "temporal/api/deployment/v1/message.proto";
+import "temporal/api/common/v1/message.proto";
+
+// Data for each worker deployment version + task queue pair.
+// This is stored in each worker deployment version (for each task queue),
+// and synced to task queue user data (for each worker deployment version).
+message TaskQueueData {
+    google.protobuf.Timestamp first_poller_time = 1;
+    google.protobuf.Timestamp last_became_current_time = 2;
+}
+
+// Local state for Worker Deployment Version
+message VersionLocalState {
+    temporal.api.deployment.v1.Deployment deployment = 1;
```

Comment:
I'm planning to also deprecate the Deployment proto. please use separate strings such as `deployment_name` and `version`/`deployment_version`.